provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/ClassPathModeExceptionCollector.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.kotlin.dsl.provider
+
+import org.gradle.groovy.scripts.ScriptSource
+import org.gradle.internal.concurrent.Stoppable
+
+
+open class ClassPathModeExceptionCollector : Stoppable {
+
+    private
+    val collection by lazy(LazyThreadSafetyMode.NONE) {
```

Comment:
:o: The possible performance advantage of instantiating a `Lazy<Collection<Exception>>` versus a `mutableListOf<Exception>` doesn't seem to pay off the additional complexity, barring special compiler support for unboxing `LazyThreadSafetyMode.NONE` lazy references. And in the end the lazy is always initialised due to `Stoppable.stop`, right?